### ISSUE TITLE
PHP Fatal Error Due to Memory Exhaustion After Plugin Update

### ISSUE THREAD

**Issue by tanjilahmed7** [polylang/polylang#1455]
Dear Support Team,

I hope this message finds you well. I am writing to report a critical issue I've encountered with Polylang after its recent update (3.6). Upon updating the plugin, my website started experiencing a fatal error which halted its operation. The specific error encountered is as follows:

**PHP Fatal error: Allowed memory size of 1073741824 bytes exhausted (tried to allocate 266240 bytes)**

Troubleshooting Attempts:

* I have tried increasing the PHP memory limit to 1GB, which is substantial, but the issue persists.
* Downgrading to the previous version is  3.5.3 of the plugin resolves the error, indicating that the issue is specific to the latest update.

I kindly request your assistance in resolving this issue. It's critical for the continued operation of my website to either have a fix for the current version or guidance on any settings I may adjust to prevent this error. If additional information or access is required to diagnose this problem, please let me know, and I'll provide it promptly.

Thank you for your attention to this matter. I look forward to your swift response and a resolution to this issue.

**CURRENT PHP VERSION : 7.4**
**DOWNGRADE PLUGIN VERSION: 3.5.3**
**LATEST PLUGIN VERSION (ERROR) : 3.6**

Best regards,
Tanjil Ahmed

**Comment by Hug0-Drelon** [comment]
Hi @tanjilahmed7,
Thanks for your feedback.
To be able to reproduce and address your issue as quickly as possible, could you provide more information?
Such as:
- WordPress version.
- If it is a multisite installation, if so whether or not Polylang is network activated.
- The full backtrace of the error.

**Comment by tanjilahmed7** [comment]
Thank you for getting back to me so quickly. 

* Wordpress version is :  6.4.3
* Yes It's multisite network website. 
* AH01071: Got error 'PHP message: PHP Fatal error: Allowed memory size of 1073741824 bytes exhausted (tried to allocate 266240 bytes) in [/var/www/vhosts/verveseniorliving.com/httpdocs/wp-content/plugins/polylang/frontend/frontend-filters-links.php](https://mydigitalchaos.ca:8443/smb/file-manager/code-editor?currentDir=%2Fhttpdocs%2Fwp-content%2Fplugins%2Fpolylang%2Ffrontend&subscriptionId=835&file=frontend-filters-links.php&redirect=%2Fsmb%2Flog-file%2Fbrowser%2Fid%2F835) on line 316PHP message:

**Comment by Chouby** [comment]
This error message is not sufficient for use to understand what happens. We need the full stack trace.

**Comment by tanjilahmed7** [comment]
> We need the full stack trace

I would appreciate it if you could explain more. Please let me know what you need from me. Unfortunately, I didn't understand what you said.

**Comment by codep0et** [comment]
> This error message is not sufficient for us to understand what happens. We need the full stack trace.

I will get back to you as soon back in the office with the full stack trace

**Comment by Chouby** [comment]
Have you some custom code using the function `switch_to_blog()`?

**Comment by Hug0-Drelon** [comment]
Thanks for the answer @codep0et,
Could you provide code snippets from what you're doing between each `switch_to_blog()`? This is for testing purpose.

**Comment by codep0et** [comment]
> Thanks for the answer @codep0et, Could you provide code snippets of what you're doing between each `switch_to_blog()`? This is for testing purpose.

Sure thing, I am using a small class to switch multiple blogs in one swoop I have put the small class below and an example code after, (get_field is ACF function to get some meta fields) : 

```
/**
 * Save and restore the current network state.
 *
 * By using this class, you can avoid unnecessary
 * switch_to_blog()-restore_current_blog()-switch_to_blog()-... excesses.
 */
class NetworkState {

        /**
         * @var int
         */
        private $site_id;

        /**
         * @var int[]
         */
        private $stack;

        /**
         * @var bool
         */
        private $switched;

        /**
         * Constructor. Sets up the properties.
         */
        public function __construct() {

                global $_wp_switched_stack, $switched;

                $this->site_id = get_current_blog_id();

                $this->stack = $_wp_switched_stack;

                $this->switched = $switched;
        }

        /**
         * Returns a new instance representing the current network state.
         *
         * @return static Network state object.
         */
        public static function create() {

                return new static();
        }

        /**
         * Restores the saved network state.
         *
         * @return void
         */
        public function restore() {

                switch_to_blog( $this->site_id );

                $GLOBALS['_wp_switched_stack'] = $this->stack;

                $GLOBALS['switched'] = $this->switched;
        }
}

add_action('wp_loaded', function(){
	$network_state = NetworkState::create();

	// MAIN SITE
	switch_to_blog(MAINSITEID);

	define('INSTAGRAM', get_field( 'instagram', 'options' ));
	define('FACEBOOK', get_field( 'facebook', 'options' ));
	define('YOUTUBE', get_field( 'YouTube', 'options' ));

	// NETWORK site
	switch_to_blog(NETWORKID);

	if ( false === as_has_scheduled_action( 'sync_pricing_cron' ) ) {
		as_schedule_recurring_action( time(), HOUR_IN_SECONDS / 4, 'sync_pricing_cron', array(), '', true );
	}

	// Network site data
	define('NEWSLETTERPAGE', get_field( 'newsletter_page', 'options' ));

	// Restore
	$network_state->restore();
});
```

**Comment by codep0et** [comment]
> Thanks for the answer @codep0et, Could you provide code snippets of what you're doing between each `switch_to_blog()`? This is for testing purpose.

Ans on other parts of the site, I am simply using a standard `switch_to_blog()` and `restore_current_blog()` like this example : 

```
function get_spontaneous_job_data(){
      switch_to_blog(NETWORKID);

    $data = [];

    $get_it = SPONTANEOUSJOB;
    if(!empty($get_it)){
        $data = [
            'id' => $get_it,
            'link' => get_the_permalink($get_it)
        ];
    }

       restore_current_blog();

    return $data;
}```

**Comment by saulirajala** [comment]
This same thing happens also in our sites. I noticed that the triggering code can be something as simple as

```php
switch_to_blog( 1 );
home_url();
restore_current_blog();
```

I did some debugging and can confirm that this happened because the current language was lost after `switch_to_blog()` as @Hug0-Drelon  mention in #1458. And I can also confirm that https://github.com/polylang/polylang/pull/1458 fixes the issue and also the original issue that https://github.com/polylang/polylang/pull/1415 was meant to fix.
